### PR TITLE
JAMES-1967 All calls to toLowerCase should use Locale.US

### PR DESCRIPTION
**Pull request by chibenwa**
I guess from the diff there was some CRLF issues

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1967-3aa23a6a-7428-4e2a-9663-4acc641b1794] Failed tests: 
[Building Java 8 james-1967-3aa23a6a-7428-4e2a-9663-4acc641b1794]   CassandraSieveRepositoryTest>AbstractSieveRepositoryTest.setActiveSwitchScriptShouldWork:197 expected:<"[Other script content]"> but was:<"[Hello World]">
```

Haha very soon this one is dead

**Comment by chibenwa** [comment]
test this please

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollection.java [hunk 1]
```diff
@@ -81,7 +82,7 @@ private Builder() {
 
         public Builder add(Field field) {
             Preconditions.checkNotNull(field);
-            String headerName = field.getName().toLowerCase();
+            String headerName = field.getName().toLowerCase(Locale.US);
```
Comment (quynhn) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollection.java]: Could we define Locale.US as default locale some where very common then it's easier to change the default locale later on?

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java [hunk 2]
```diff
@@ -370,7 +371,7 @@ protected Logger getLogger() {
      *            the domain
      * @param mapping
      *            the mapping
-     * @throws InvalidMappingException
+     * @throws RecipientRewriteTableException
```
Comment (quynhn) [inline server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java]: In separated commit, no?

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java [hunk 3]
```diff
@@ -383,7 +384,7 @@ protected Logger getLogger() {
...
```
Comment (quynhn) [inline server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java]: Idem

Comment (chibenwa) [inline server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java]: Arf, if you want

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollection.java (hunk 1 above)
Comment (chibenwa) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollection.java]: Nope. Internationalization is not supported anywhere else. It's not part of this task.

This task is "get a predictable behaviour on toLowerCase". It's very different from "i8n everywhere"

Comment (quynhn) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollection.java]: Alright

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**